data/mods.json

```diff
@@ -494,5 +494,27 @@
     "size": 2949,
     "url": "https://github.com/Entity-y/OG_Drugs/archive/main.zip",
     "homepage": "https://discourse.cataclysmdda.org/t/original-drug-names/25335"
+  },
+  {
+    "type": "direct_download",
+    "ident": [ "Kenan2000s_modpack" ],

```

Comment:
I think you need to put identifiers of all mods in the archive, e.g.:

https://github.com/remyroy/CDDA-Game-Launcher/blob/3eb95c9149f349fc600a69bc2a2152a6bc4843e3/data/mods.json#L281

Reply:
I don't think I'll ever be able to do that, as I'm in college. I would definitely appreciate if someone else could though. I personally don't think it's necessary, as there are so many mods in the modpack that it makes it almost pointless to show all of them.

Reply:
It is not about display - it is a requirement, so it could work properly.

Number of mods in the modpack does not matter as you do not need to write down ids manually. You can write some script to extract mod idents or alternatively do a file search to gather mod identifiers and manual formatting after that - it takes just a couple of minutes, e.g. for vanilla mods:

![image](https://user-images.githubusercontent.com/16213433/108345452-54dfaf00-71ef-11eb-80b2-5b37c81e067e.png)

```
Search ""id"" (40 hits in 39 files of 39 searched)
  C:\Projects\Cataclysm-DDA\data\mods\Aftershock\modinfo.json (1 hit)
	Line 4:     "id": "aftershock",
  C:\Projects\Cataclysm-DDA\data\mods\alt_map_key\modinfo.json (1 hit)
	Line 4:     "id": "alt_map_key",
  C:\Projects\Cataclysm-DDA\data\mods\BlazeIndustries\modinfo.json (1 hit)
	Line 4:     "id": "blazeindustries",
  C:\Projects\Cataclysm-DDA\data\mods\cbm_slots\modinfo.json (1 hit)
	Line 4:     "id": "cbm_slots",
  C:\Projects\Cataclysm-DDA\data\mods\Chibi_Ultica\modinfo.json (1 hit)
	Line 4:     "id": "Chibi_Ultica",
  C:\Projects\Cataclysm-DDA\data\mods\classic_zombies\modinfo.json (1 hit)
	Line 4:     "id": "classic_zombies",
  C:\Projects\Cataclysm-DDA\data\mods\CrazyCataclysm\modinfo.json (1 hit)
	Line 4:     "id": "crazy_cataclysm",
  C:\Projects\Cataclysm-DDA\data\mods\CRT_EXPANSION\modinfo.json (1 hit)
	Line 4:     "id": "crt_expansion",
  C:\Projects\Cataclysm-DDA\data\mods\Dark-Skies-Above\modinfo.json (1 hit)
	Line 4:     "id": "darkskies",
  C:\Projects\Cataclysm-DDA\data\mods\dda\modinfo.json (1 hit)
	Line 4:     "id": "dda",
  C:\Projects\Cataclysm-DDA\data\mods\desert_region\modinfo.json (1 hit)
	Line 4:     "id": "desertpack",
  C:\Projects\Cataclysm-DDA\data\mods\DinoMod\modinfo.json (1 hit)
	Line 4:     "id": "DinoMod",
  C:\Projects\Cataclysm-DDA\data\mods\Fuji_Mil_Prof\modinfo.json (1 hit)
	Line 4:     "id": "fuji_mpp",
  C:\Projects\Cataclysm-DDA\data\mods\Fuji_Structures\modinfo.json (1 hit)
	Line 4:     "id": "FujiStruct",
  C:\Projects\Cataclysm-DDA\data\mods\Generic_Guns\modinfo.json (1 hit)
	Line 4:     "id": "generic_guns",
  C:\Projects\Cataclysm-DDA\data\mods\Graphical_Overmap\modinfo.json (1 hit)
	Line 4:     "id": "Graphical_Overmap",
  C:\Projects\Cataclysm-DDA\data\mods\Graphical_Overmap_FujiStruct\modinfo.json (1 hit)
	Line 4:     "id": "Graphical_Overmap_Fujistruct",
  C:\Projects\Cataclysm-DDA\data\mods\Graphical_Overmap_Magiclysm\modinfo.json (1 hit)
	Line 4:     "id": "Graphical_Overmap_Magiclysm",
  C:\Projects\Cataclysm-DDA\data\mods\Graphical_Overmap_More_Locations\modinfo.json (1 hit)
	Line 4:     "id": "Graphical_Overmap_More_Locations",
  C:\Projects\Cataclysm-DDA\data\mods\Graphical_Overmap_Urban_Development\modinfo.json (1 hit)
	Line 4:     "id": "Graphical_Overmap_Urban_Development",
  C:\Projects\Cataclysm-DDA\data\mods\Magiclysm\modinfo.json (2 hits)
	Line 4:     "id": "magiclysm",
	Line 14:     "id": "spellcraft",
  C:\Projects\Cataclysm-DDA\data\mods\MMA\modinfo.json (1 hit)
	Line 4:     "id": "MMA",
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\modinfo.json (1 hit)
	Line 4:     "id": "more_locations",
  C:\Projects\Cataclysm-DDA\data\mods\Mutant_NPCs\modinfo.json (1 hit)
	Line 4:     "id": "mutant_npcs",
  C:\Projects\Cataclysm-DDA\data\mods\My_Sweet_Cataclysm\modinfo.json (1 hit)
	Line 4:     "id": "my_sweet_cataclysm",
  C:\Projects\Cataclysm-DDA\data\mods\National_Guard_Camp\modinfo.json (1 hit)
	Line 4:     "id": "national_guard_camp",
  C:\Projects\Cataclysm-DDA\data\mods\No_Fungi\modinfo.json (1 hit)
	Line 4:     "id": "No_Fungi",
  C:\Projects\Cataclysm-DDA\data\mods\No_NPC_Food\modinfo.json (1 hit)
	Line 4:     "id": "no_npc_food",
  C:\Projects\Cataclysm-DDA\data\mods\No_Rail_Stations\modinfo.json (1 hit)
	Line 4:     "id": "No_Rail_Stations",
  C:\Projects\Cataclysm-DDA\data\mods\Only_Wildlife\modinfo.json (1 hit)
	Line 4:     "id": "Only_Wildlife",
  C:\Projects\Cataclysm-DDA\data\mods\package_bionic_professions\modinfo.json (1 hit)
	Line 4:     "id": "package_bionic_professions",
  C:\Projects\Cataclysm-DDA\data\mods\rural_biome\modinfo.json (1 hit)
	Line 4:     "id": "ruralbiome",
  C:\Projects\Cataclysm-DDA\data\mods\sees_player_hitbutton\modinfo.json (1 hit)
	Line 4:     "id": "sees_player_hitbutton",
  C:\Projects\Cataclysm-DDA\data\mods\sees_player_retro\modinfo.json (1 hit)
	Line 4:     "id": "sees_player_retro",
  C:\Projects\Cataclysm-DDA\data\mods\speedydex\modinfo.json (1 hit)
	Line 4:     "id": "speedydex",
  C:\Projects\Cataclysm-DDA\data\mods\StatsThroughSkills\modinfo.json (1 hit)
	Line 4:     "id": "StatsThroughSkills",
  C:\Projects\Cataclysm-DDA\data\mods\stats_through_kills\modinfo.json (1 hit)
	Line 4:     "id": "stats_through_kills",
  C:\Projects\Cataclysm-DDA\data\mods\TEST_DATA\modinfo.json (1 hit)
	Line 4:     "id": "test_data",
  C:\Projects\Cataclysm-DDA\data\mods\Urban_Development\modinfo.json (1 hit)
	Line 4:     "id": "Urban_Development",
```

![image](https://user-images.githubusercontent.com/16213433/108345584-80fb3000-71ef-11eb-8570-5845492ad245.png)

![image](https://user-images.githubusercontent.com/16213433/108345766-b3a52880-71ef-11eb-819c-56f3ec337394.png)

```
"spellcraft",
"Chibi_Ultica",
"DinoMod",
"FujiStruct",
"Graphical_Overmap",
"Graphical_Overmap_Fujistruct",
"Graphical_Overmap_Magiclysm",
"Graphical_Overmap_More_Locations",
"Graphical_Overmap_Urban_Development",
"MMA",
"No_Fungi",
"No_Rail_Stations",
"Only_Wildlife",
"StatsThroughSkills",
"Urban_Development",
"aftershock",
"alt_map_key",
"blazeindustries",
"cbm_slots",
"classic_zombies",
"crazy_cataclysm",
"crt_expansion",
"darkskies",
"dda",
"desertpack",
"fuji_mpp",
"generic_guns",
"magiclysm",
"more_locations",
"mutant_npcs",
"my_sweet_cataclysm",
"national_guard_camp",
"no_npc_food",
"package_bionic_professions",
"ruralbiome",
"sees_player_hitbutton",
"sees_player_retro",
"speedydex",
"stats_through_kills",
"test_data",
```

Reply:
If I messed anything up, please tell me.